convert/src/vertex_to_bd/vertex_to_badgerdoc_use_case.py

```diff
@@ -1,53 +1,72 @@
+
+from src.config import DEFAULT_PAGE_BORDER_OFFSET, settings
+from src.logger import get_logger
+from src.vertex_to_bd.badgerdoc_format.annotation_converter import (
+    AnnotationConverter,
+)
+from src.vertex_to_bd.badgerdoc_format.badgerdoc_format import BadgerdocFormat
+from src.vertex_to_bd.badgerdoc_format.pdf_renderer import PDFRenderer
+from src.vertex_to_bd.badgerdoc_format.plain_text_converter import (
     TextToBadgerdocTokensConverter,
 )
-from .models.vertex_models import LabelStudioModel, S3Path
+from src.vertex_to_bd.models.vertex_models import LabelStudioModel, S3Path
+
+LOGGER = get_logger(__file__)
 
 
 class VertexToBDConvertUseCase:
     def __init__(
         self,
         s3_client,
+        current_tenant: str,
+        token_data: TenantData,
+        s3_input_annotation: S3Path,
+        s3_output_bucket: str,
+        s3_output_file_id: int
     ) -> None:
         page_border_offset = DEFAULT_PAGE_BORDER_OFFSET
         self.plain_text_converter = TextToBadgerdocTokensConverter(
             page_border_offset=page_border_offset
         )
         self.pdf_renderer = PDFRenderer(page_border_offset=page_border_offset)
         self.annotation_converter = AnnotationConverter()
-
         self.badgerdoc_format = BadgerdocFormat()
         self.s3_client = s3_client
 
-    def execute(
-        self,
-        s3_input_annotation: S3Path,
-        s3_output_pdf: S3Path,
-        s3_output_tokens: S3Path,
-        s3_output_annotations: S3Path,
-    ) -> None:
+        self.current_tenant = current_tenant
+        self.token_data = token_data
+        self.request_headers = {
+            "X-Current-Tenant": self.current_tenant,
+            "Authorization": f"Bearer {self.token_data.token}",
+        }
+        LOGGER.info(f"{self.request_headers=}")
+        self.s3_input_annotation = s3_input_annotation
+        self.s3_output_bucket = s3_output_bucket
+        self.s3_output_file_id = s3_output_file_id
+
+    def execute(self) -> None:
         labelstudio_format = self.download_labelstudio_from_s3(
-            s3_input_annotation
+            self.s3_input_annotation
         )
+        LOGGER.info(f"label studio format {labelstudio_format=}")
         self.badgerdoc_format.convert_from_labelstudio(labelstudio_format)
-        self.upload_badgerdoc_to_s3(
-            s3_output_tokens,
-            s3_output_pdf,
-            s3_output_annotations,
-        )
+        LOGGER.info("Tokens and annotations are converted")
+        importjob_id_created = self.import_annotations_to_annotation_microservice()
+        self.upload_badgerdoc_format_to_s3(importjob_id_created)
+        LOGGER.info("Tokens and annotations uploaded")
+
 
     def download_labelstudio_from_s3(
         self,
         s3_input_annotation: S3Path,
     ) -> LabelStudioModel:
-        with tempfile.TemporaryDirectory() as tmp_dirname:
+        with tempfile.TemporaryDirectory(dir=".") as tmp_dirname:
             tmp_dirname = Path(tmp_dirname)
```

Comment:
To make code more maintainable I would not assing different types to the same var

Reply:
I've missed this one. dir="." was only used for testing purposes and I deleted it in production code.

Reply:
As I get the question is about "tmp_dirname = Path(tmp_dirname)"